vllm/entrypoints/llm.py

```diff
@@ -438,7 +502,31 @@ def generate(
         )
 
         outputs = self._run_engine(use_tqdm=use_tqdm)
-        return self.engine_class.validate_outputs(outputs, RequestOutput)
+        model_outputs = self.engine_class.validate_outputs(outputs, RequestOutput)
+
+        if (
+            self.io_processor is not None
+            and self.io_processor.plugin_type != IOProcessorPluginType.INPUT_ONLY
+        ):
+            # get the post-processed model outputs
+            processed_outputs = self.io_processor.post_process(
+                model_output=model_outputs
+            )
```

Comment:
I would do exactly the same for the pre-processing part, figuring out if the plugin should be applied in this function.